python/src/worker/tasks/marker_heatmap.py

```diff
@@ -27,14 +27,14 @@ def _format_result(self, result):
         backoff.expo, requests.exceptions.RequestException, max_time=30
     )
     def compute(self):
-        request = {"nGenes": self.task_def["nGenes"], "type":self.task_def["type"], "config":self.task_def["config"]}
+        request = {"nGenes": self.task_def["nGenes"]}
         
-        typeOfSets = self.task_def["typeOfSets"]
+        cellSetKey = self.task_def["cellSetKey"]
 
         cellSets = get_cell_sets(self.experiment_id)
 
         for set in cellSets:
-            if(set["key"]==typeOfSets):
+            if(set["key"]==cellSetKey):
                 cellSets = set
                 break
```

Comment:
where do you actually use this `cellSetKey`? It still looks unnecessary to me

Reply:
This `cellSetKey` is used to access the key containing cell set data that will be used to generate the marker heatmap. It is not of much use right now because `louvain` is the only option. Looking at the clustering method dropdown, we will support `leiden` and `SLM`.  Leaving this check here will allow us to easily switch between the clustering methods once they're implemented.

Reply:
Yes, I think you are right, we can leave it